Subscriptions view

TomHennen opened the pull request:
finishes the work of #1624 

Made some of the changes recommended in that PR.

@mfietz I'd appreciate your input.  I think the way the images and the placeholders are working isn't quite right and I'd like your ideas.

TomHennen commented:
![device-2016-03-25-135959](https://cloud.githubusercontent.com/assets/5216560/14050387/eed36a9e-f291-11e5-89ee-0719ada451e0.png)

![device-2016-03-25-140052](https://cloud.githubusercontent.com/assets/5216560/14050396/036d0d98-f292-11e5-87a5-5e5bbe878308.png)

mfietz commented:
Few things that bugged me:
* Action bar looks really empty: Possible actions would be search (either episodes or only show podcasts that contain the search string), refresh, filter...
* No long click action: I would assume that I get at least the same context menu actions as in the drawer
* There should be a link to add feed: Either the last cell should be dedicated to that (big plus sign, title add feed) or as the "there are no items in this list" alternative.
* When I click on a subscription and then push the back button, I would assume it takes me back to the subscription overview. Just doesn't feel right. Instead, the app is closed...
* The GridView cells(?) don't seem to know their height until the image is loaded. Looks really weird while loading has not completed.
* Should have some padding and the margin between the images also could be a little bigger
* No counter: I think I should see the same numbers that I also see in the drawer. Else the order of the subscriptions just seems random.

mfietz commented:
For counter: https://github.com/shts/TriangleLabelView

keunes commented:
@mfietz: I personally find the label originally proposed in #836 way more elegant than the triangle, I must say.

Matth7878 commented:
I'm in favor of using triangles : it looks nicer to me and it stands out more.

TomHennen commented:
> Action bar looks really empty: Possible actions would be search (either episodes or only show podcasts that contain the search string), refresh, filter...

I don't think I like any of these. Would take a fair bit of work to implement.  What I do think we should add is 'Add Subscription'.  We might even remove it from the Nav Drawer.

> No long click action: I would assume that I get at least the same context menu actions as in the drawer

Added one that brings up the same as in the drawer.

> There should be a link to add feed: Either the last cell should be dedicated to that (big plus sign, title add feed) or as the "there are no items in this list" alternative.

Do you think this is better than just putting in the action bar?

> When I click on a subscription and then push the back button, I would assume it takes me back to the subscription overview. Just doesn't feel right. Instead, the app is closed...

This is how all the other items in the nav bar work.  There was an issue where someone wanted it to bring the nav bar back.  You could do that...

> The GridView cells(?) don't seem to know their height until the image is loaded. Looks really weird while loading has not completed.

I think I've resolved this.

> Should have some padding and the margin between the images also could be a little bigger

How could the images be bigger? Only 2 columns?

> No counter: I think I should see the same numbers that I also see in the drawer. Else the order of the subscriptions just seems random.

Added counter.  I went with triangles over the circle because the circle didn't have room for 3 digit numbers.

![subscriptions](https://cloud.githubusercontent.com/assets/5216560/14061676/8093cdb4-f35c-11e5-8640-3ce5e3081e44.png)

I'm reticent to do any more on this PR.  My main goal was to get the other contribution in to an acceptable state.  Once this is merged it should be easy to add other features (filtering, add feed, etc...)

Current thoughts?

mfietz commented:
> We might even remove it from the Nav Drawer.

Don't think this is a good idea. At least as long as we don't have "Add feed" on any empty list message where it makes sense (all episodes, new episodes, queue, subscriptions)

> Do you think this is better than just putting in the action bar?

Having add feed in the action bar sounds like bad UX. Even the FAB would be better. But as this thing is pretty ugly and always on top of the thing you are most interested in, I think an extra cell is much better.
(Probably in a later PR)

> This is how all the other items in the nav bar work. There was an issue where someone wanted it to bring the nav bar back. You could do that... 

The problem is not that the app closed. My problem is that I would expect to be taken back to subscriptions. When I go from Feed to an episode, back takes me back to the feed. When I go from subscriptions to a feed, there is no going back. But I would assume that, it fells just natural.

> Only 2 columns?
I think 3 columns is fine.

mfietz commented:
Also, up for discussion: I think subscriptions should be at the top of the drawer, even before queue. Also think it needs another icon, I propose https://design.google.com/icons/#ic_subscriptions
(Subscriptions is kind of like "Home")

keunes commented:
> Action bar looks really empty
> > What I do think we should add is 'Add Subscription'. We might even remove it from the Nav Drawer.

> Don't think this is a good idea. At least as long as we don't have "Add feed" on any empty list message where it makes sense

Agree w/ remove from Nav Drawer. i see an [empty state screen](https://www.google.com/design/spec/patterns/empty-states.html#empty-states-displaying-empty-states) as 'great to have', but not as a requirement to remove 'add episode' from the Nav Drawer. An action bar button or cell with big plus to add episodes would be.

>  Once this is merged it should be easy to add other features (filtering, add feed, etc...)

One of those IMHO would be a sort function (alphabetical OR # of episodes (as per settings))

> When I go from Feed to an episode, back takes me back to the feed. When I go from subscriptions to a feed, there is no going back. But I would assume that

Agree

> subscriptions should be at the top of the drawer, even before queue

Top of the drawer: yes. Before the queue: absolutely not. The queue is the epicentre of AP, not the feeds. We can't even use feeds as a 'playlist' atm. AP is about listening to podcasts (which most people do via the queue), not about managing subscriptions (or whatever the subscription view allows for. The queue should be the 'home'.

> Also think it needs another icon, I propose https://design.google.com/icons/#ic_subscriptions

You want to do some YouTube product placement? o_O AP's main purpose isn't even to play video :P I also have my doubts about the 'folder' icon. Maybe we could follow Pocket Cast's example and show a grid: https://design.google.com/icons/#ic_apps (or we can ask the folks over at https://github.com/Templarian/MaterialDesign to make an adapted version (to turn 90°) of https://design.google.com/icons/#ic_view_comfy)

TomHennen commented:
Last update of any consequence.

* Added 'Add Podcast' to Subscriptions view
* Moved 'Subscriptions' up in the nav drawer. Under Episodes but before Downloads.
* Fixed some minor bugs.

Comments welcome.

![subscription_with_add](https://cloud.githubusercontent.com/assets/5216560/14067207/508a0472-f42d-11e5-8efb-807a72ff2713.png)

mfietz commented:
If you don't want to change anything else yourself, I think this is ready to be merged.

TomHennen commented:
What is left to do? Some action for the action bar?

keunes commented:
I would expect the 'Add podcast' item at the bottom/end of the list btw. Seems like a waste of space in the 'above the fold' section.

And with the risk of nagging: it would be nice if text justification would be centre. And also afraid off the Dutch translation being too long to fit on one line ("podcast toevoegen "), so hope its flexibly set up.

Matth7878 commented:
Tried with 1.6.0.0 and my feelings are :
- We can choose whether to display feeds in navigation drawer and we can do that for subscription view but not the same way => there should be a "subscriptions list" in "change navigation drawer items" to make it coherent with others items from navigation drawer. (didn't notice it in mock-up)
- I think subscriptions button should be at the bottom just before feeds list when both are displayed. Moving it up is ok when feed list is not displayed but they really feel apart when they're both in navigation drawer.
- add podcast button in subscriptions view : like @keunes I would prefer it as the last item. Or as a button in action bar to make it more visible for people with a lot of podcasts but not as first item. Beside since subscriptions button is displayed by default we should not have add podcast in navigation drawer visible as default. (Letting those who want it there to turn it on)
